File: src/RealWorld.Application/DTOs/UserData.cs
namespace RealWorld.Application.DTOs;

public class UserData
{
    public string Id { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Username { get; set; } = string.Empty;
    public string Bio { get; set; } = string.Empty;
    public string Image { get; set; } = string.Empty;

    public UserData() { }

    public UserData(string id, string email, string username, string bio, string image)
    {
        Id = id;
        Email = email;
        Username = username;
        Bio = bio;
        Image = image;
    }
}

File: src/main/java/io/spring/application/data/UserData.java
package io.spring.application.data;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/** Read-model DTO for user account data (excludes password). */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserData {
  private String id;
  private String email;
  private String username;
  private String bio;
  private String image;
}
